sycl/doc/SYCLPluginInterface.md

```diff
@@ -0,0 +1,178 @@
+*__Example of PI non OpenCL-based APIs__*
+
+Selects the most appropriate device image based on runtime
+information and the image characteristics
+```
+cl_int piSelectDeviceImage(pi_context ctx, pi_device_image **images,
+                             cl_uint num_images,
+                             pi_device_image **selected_image);
+```
+
+PI also defines few types and string tags to describe a device binary image.
+Those are used to communicate to plugins information about the images where it
+is needed, currently only in the ```piSelectDeviceImage``` function. The main
+type is ```pi_device_image```, whose detailed description can also be found in
+the header.  The layout of this type strictly matches the layout of the corresponding
+device image descriptor type defined in the ```clang-offload-wrapper``` tool
+which wraps device binaries into a host object for further linkage. The wrapped
+images reside inside this descriptor in a data section.
+
+### The Extended PI APIs
+
+These are APIs needed to implement SYCL-to-OpenCL interoperability. Only
+the OpenCL plugin will provide these APIs, generic plugins won't, because
+SYCL runtime will cast PI object pointers to/from OpenCL object pointers when
+using result of/invoking the extended APIs in interoperability scenario. These
+APIs match semantics of the corresponding OpenCL APIs exactly.
+
+The full list of such APIs can be found in the
+[pi.h](../include/CL/sycl/detail/pi.h) header. One example is:
+
+    piOpenCLCreateProgramWithSource
+ 
+### Extension mechanism
+
+This section describes a mechanism for SYCL or other runtimes to detect
+availability of and obtain interfaces beyond those defined by the PI dispatch.
+The ```piGetExtensionFunctionAddressForPlatform``` function can be used
+to find a specific function pointer and call it to get an additional table of
+function pointers (if they are provided by implementation), for example:
+
+
+    getOMPDispatchTable =
+      (getOMPDispatchTable_type)
+      piGetExtensionFunctionAddressForPlatform(platf_id,_
+                                         “getOMPDispatchTable”);
+    OMPDispatchTable = getOMPDispatchTable(OMPVersion);
+
+### Plugin access to SYCL runtime interface
+
+The PI design allows plugins to get access to some of the SYCL runtime
+functionality, such as global resource acquisition and release, asynchronous
+completion signaling (TBD). The same approach as for obtaining additional
+interfaces can be used here. For example:
+
+Types defined by PI:
+
+    enum ErrorCode {…};
+    …
+    struct PIDisptachTable {
+      CFunctionPtrType1 function1;
+      CFunctionPtrType2 function2;
+    …
+    };
+    typedef ErrorCode
+      (*PIDisptachTableSetterF)(PIDisptachTable *sycl_rt_intf);
+
+SYCL RT code:
+
+    static PIDisptachTable sycl_rt_intf;
+    …
+    PIDisptachTableSetterF f =
+      (SYCLRtInterfaceSetterF)clGetExtensionFunctionAddressForPlatform(
+        platf_id, “setSYCLRtInterface”);
+    if (f) f(&sycl_rt_intf);
```

Comment:
The example is not very clear, what is it trying to do? and what is "setSYCLRtInterface" doing?
As @smaslov-intel  suggests above, its better to leave this as a TBD, and after there is more implementation experience, the documentation can be updated - this shouldn't be a blocker for this PR to come through.

Reply:
OK